network/eos/eos_eapi.py

```diff
@@ -35,259 +35,335 @@
+    commands.append('qos dscp %s' % module.params['qos'])
+
+def get_config(module):
+    contents = module.params['config']
+    if not contents:
+        cmd = 'show running-config all section management api http-commands'
+        contents = module.cli([cmd])
+    config = NetworkConfig(indent=3, contents=contents[0])
+    return config
+
+def load_config(module, commands, result):
+    session = 'ansible_%s' % int(time.time())
+    commit = not module.check_mode
+
+    diff = module.config.load_config(commands, session=session, commit=commit)
+
+    # once the configuration is done, remove the config session and
+    # remove the session name from the result
+    module.cli(['no configure session %s' % session])
+
+    result['diff'] = dict(prepared=diff)
+    result['changed'] = diff is not None
+
+def load(module, commands, result):
+    candidate = NetworkConfig(indent=3)
+    candidate.add(commands, parents=['management api http-commands'])
+
+    config = get_config(module)
+    configobjs = candidate.difference(config)
+
+    if configobjs:
+        commands = dumps(configobjs, 'commands').split('\n')
+        result['updates'] = commands
+        load_config(module, commands, result)
+
+def clean_result(result):
+    # strip out any keys that have two leading and two trailing
+    # underscore characters
+    for key in result.keys():
+        if PRIVATE_KEYS_RE.match(key):
+            del result[key]
+
+def collect_facts(module, result):
+    resp = module.cli(['show management api http-commands'], output='json')
+    facts = dict(eos_eapi_urls=dict())
+    for each in resp[0]['urls']:
+        intf, url = each.split(' : ')
+        key = str(intf).strip()
+        if  key not in facts['eos_eapi_urls']:
+            facts['eos_eapi_urls'][key] = list()
+        facts['eos_eapi_urls'][key].append(str(url).strip())
+    result['ansible_facts'] = facts
 
-def check_version(module):
-    config = module.from_json(module.execute(['show version | json'])[0])
-    versions = config['version'].split('.')
-    if int(versions[0]) < 4 or int(versions[1]) < 12:
-        module.fail_json(msg="Device version {0} does not support eAPI. eAPI was introduced in EOS 4.12.")
 
 def main():
     """ main entry point for module execution
     """
 
     argument_spec = dict(
-        state=dict(default='started', choices=['stopped','started']),
-        http_port=dict(default=80, type='int'),
-        https_port=dict(default=443, type='int'),
-        local_http_port=dict(default=8080, type='int'),
-        http=dict(aliases=['enable_http'], default=True, type='bool'),
-        https=dict(aliases=['enable_https'], default=True, type='bool'),
+        http=dict(aliases=['enable_http'], default=False, type='bool', setter='set_protocol_http'),
+        http_port=dict(default=80, type='int', setter='set_protocol_http'),
+
+        https=dict(aliases=['enable_https'], default=True, type='bool', setter='set_protocol_https'),
```

Comment:
For some other modules (in other network appliances) we have `https` `default=False`
Is their a reason we don't have https default to yes across the board?